ddtrace/tracer/payload.go

```diff
@@ -102,7 +102,7 @@ const (
 // updateHeader updates the payload header based on the number of items currently
 // present in the stream.
 func (p *payload) updateHeader() {
-	n := atomic.LoadUint64(&p.count)
+	n := uint64(atomic.LoadUint32(&p.count))
```

Comment:
What's the reason behind converting this to a `uint64` then calling `binary.BigEndian.PutUint64(p.header, n)`, instead of leaving it a `uint32` and just using `binary.BigEndian.PutUint32(p.header, n)`? Will that mess with the header in some way?

Reply:
This is just to keep the msgpack munging we already have below. It's probable that we could make it work with a 32-bit integer, but there's no real performance penalty this way and the 64-bit code has been working forever. No need to change it now.